regression/cbmc/static_assert/static_assert1.c

```diff
@@ -0,0 +1,17 @@
+// C23 introduces the "static_assert" keyword.
+
+struct S
+{
+  // Visual Studio does not support static_assert in compound bodies.
+#ifndef _MSC_VER
+  static_assert(1, "in struct");
+#endif
+  int x;
+} asd;
+
+static_assert(1, "global scope");
+
+int main()
+{
+  static_assert(1, "in function");
+}
```

Comment:
Shouldn't this go in the `ansi-c` regression test folder instead of `cbmc`?

Reply:
Yes, that was my first attempt. But we run various compilers over that folder, and they don't do C23.